In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import warnings
warnings.filterwarnings("ignore")

In [2]:
tmdb_movies=pd.read_csv("C:/Users/ADMIN/Desktop/tmdb_movies.csv")

this is used to insert the daset file to be used during the analysis 

In [3]:
tmdb_movies.head(10)

id    imdb_id  popularity     budget     revenue  \
0  135397  tt0369610   32.985763  150000000  1513528810   
1   76341  tt1392190   28.419936  150000000   378436354   
2  262500  tt2908446   13.112507  110000000   295238201   
3  140607  tt2488496   11.173104  200000000  2068178225   
4  168259  tt2820852    9.335014  190000000  1506249360   
5  281957  tt1663202    9.110700  135000000   532950503   
6   87101  tt1340138    8.654359  155000000   440603537   
7  286217  tt3659388    7.667400  108000000   595380321   
8  211672  tt2293640    7.404165   74000000  1156730962   
9  150540  tt2096673    6.326804  175000000   853708609   

                 original_title  \
0                Jurassic World   
1            Mad Max: Fury Road   
2                     Insurgent   
3  Star Wars: The Force Awakens   
4                     Furious 7   
5                  The Revenant   
6            Terminator Genisys   
7                   The Martian   
8                       Minions   
9                    Inside Out   

                                                cast  \
0  Chris Pratt|Bryce Dallas Howard|Irrfan Khan|Vi...   
1  Tom Hardy|Charlize Theron|Hugh Keays-Byrne|Nic...   
2  Shailene Woodley|Theo James|Kate Winslet|Ansel...   
3  Harrison Ford|Mark Hamill|Carrie Fisher|Adam D...   
4  Vin Diesel|Paul Walker|Jason Statham|Michelle ...   
5  Leonardo DiCaprio|Tom Hardy|Will Poulter|Domhn...   
6  Arnold Schwarzenegger|Jason Clarke|Emilia Clar...   
7  Matt Damon|Jessica Chastain|Kristen Wiig|Jeff ...   
8  Sandra Bullock|Jon Hamm|Michael Keaton|Allison...   
9  Amy Poehler|Phyllis Smith|Richard Kind|Bill Ha...   

                                            homepage  \
0                      http://www.jurassicworld.com/   
1                        http://www.madmaxmovie.com/   
2     http://www.thedivergentseries.movie/#insurgent   
3  http://www.starwars.com/films/star-wars-episod...   
4                           http://www.furious7.com/   
5       http://www.foxmovies.com/movies/the-revenant   
6                    http://www.terminatormovie.com/   
7        http://www.foxmovies.com/movies/the-martian   
8                       http://www.minionsmovie.com/   
9                http://movies.disney.com/inside-out   

                         director  \
0                 Colin Trevorrow   
1                   George Miller   
2                Robert Schwentke   
3                     J.J. Abrams   
4                       James Wan   
5  Alejandro González Iñárritu   
6                     Alan Taylor   
7                    Ridley Scott   
8        Kyle Balda|Pierre Coffin   
9                     Pete Docter   

                                           tagline  ...  \
0                                The park is open.  ...   
1                               What a Lovely Day.  ...   
2                       One Choice Can Destroy You  ...   
3                    Every generation has a story.  ...   
4                              Vengeance Hits Home  ...   
5  (n. One who has returned, as if from the dead.)  ...   
6                                 Reset the future  ...   
7                                   Bring Him Home  ...   
8     Before Gru, they had a history of bad bosses  ...   
9         Meet the little voices inside your head.  ...   

                                            overview runtime  \
0  Twenty-two years after the events of Jurassic ...     124   
1  An apocalyptic story set in the furthest reach...     120   
2  Beatrice Prior must confront her inner demons ...     119   
3  Thirty years after defeating the Galactic Empi...     136   
4  Deckard Shaw seeks revenge against Dominic Tor...     137   
5  In the 1820s, a frontiersman, Hugh Glass, sets...     156   
6  The year is 2029. John Connor, leader of the r...     125   
7  During a manned mission to Mars, Astronaut Mar...     141   
8  Minions Stuart, Kevin and Bob are recruited by...      91   
9  Growing up can be a bumpy road, and 

In [4]:
tmdb_movies.tail(5)

id    imdb_id  popularity  budget  revenue  \
10861     21  tt0060371    0.080598       0        0   
10862  20379  tt0060472    0.065543       0        0   
10863  39768  tt0060161    0.065141       0        0   
10864  21449  tt0061177    0.064317       0        0   
10865  22293  tt0060666    0.035919   19000        0   

                 original_title  \
10861        The Endless Summer   
10862                Grand Prix   
10863       Beregis Avtomobilya   
10864    What's Up, Tiger Lily?   
10865  Manos: The Hands of Fate   

                                                    cast homepage  \
10861  Michael Hynson|Robert August|Lord 'Tally Ho' B...      NaN   
10862  James Garner|Eva Marie Saint|Yves Montand|Tosh...      NaN   
10863  Innokentiy Smoktunovskiy|Oleg Efremov|Georgi Z...      NaN   
10864  Tatsuya Mihashi|Akiko Wakabayashi|Mie Hama|Joh...      NaN   
10865  Harold P. Warren|Tom Neyman|John Reynolds|Dian...      NaN   

                 director                                            tagline  \
10861         Bruce Brown                                                NaN   
10862  John Frankenheimer  Cinerama sweeps YOU into a drama of speed and ...   
10863      Eldar Ryazanov                                                NaN   
10864         Woody Allen                          WOODY ALLEN STRIKES BACK!   
10865    Harold P. Warren       It's Shocking! It's Beyond Your Imagination!   

       ...                                           overview runtime  \
10861  ...  The Endless Summer, by Bruce Brown, is one of ...      95   
10862  ...  Grand Prix driver Pete Aron is fired by his te...     176   
10863  ...  An insurance agent who moonlights as a carthie...      94   
10864  ...  In comic Woody Allen's film debut, he took the...      80   
10865  ...  A family gets lost on the road and stumbles up...      74   

                       genres  \
10861             Documentary   
10862  Action|Adventure|Drama   
10863          Mystery|Comedy   
10864           Action|Comedy   
10865                  Horror   

                                    production_companies release_date  \
10861                                  Bruce Brown Films      6/15/66   
10862  Cherokee Productions|Joel Productions|Douglas ...     12/21/66   
10863                                            Mosfilm   01/01/1966   
10864                            Benedict Pictures Corp.   11/02/1966   
10865                                          Norm-Iris     11/15/66   

      vote_count  vote_average  release_year   budget_adj  revenue_adj  
10861         11           7.4          1966       0.0000          0.0  
10862         20           5.7          1966       0.0000          0.0  
10863         11           6.5          1966       0.0000          0.0  
10864         22           5.4          1966       0.0000          0.0  
10865         15           1.5          1966  127642.2792          0.0  

[5 rows x 21 columns]

In [5]:
tmdb_movies.shape

(10866, 21)

this attribute returns the dimensions of the dataset 

In [6]:
tmdb_movies.columns

Index(['id', 'imdb_id', 'popularity', 'budget', 'revenue', 'original_title',
       'cast', 'homepage', 'director', 'tagline', 'keywords', 'overview',
       'runtime', 'genres', 'production_companies', 'release_date',
       'vote_count', 'vote_average', 'release_year', 'budget_adj',
       'revenue_adj'],
      dtype='object')

In [7]:
tmdb_movies.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10866 entries, 0 to 10865
Data columns (total 21 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   id                    10866 non-null  int64  
 1   imdb_id               10856 non-null  object 
 2   popularity            10866 non-null  float64
 3   budget                10866 non-null  int64  
 4   revenue               10866 non-null  int64  
 5   original_title        10866 non-null  object 
 6   cast                  10790 non-null  object 
 7   homepage              2936 non-null   object 
 8   director              10822 non-null  object 
 9   tagline               8042 non-null   object 
 10  keywords              9373 non-null   object 
 11  overview              10862 non-null  object 
 12  runtime               10866 non-null  int64  
 13  genres                10843 non-null  object 
 14  production_companies  9836 non-null   object 
 15  release_date       

this method prints information about the DataFrame.
there are 10866 entrie with a total of 21 columns.
it also indicates the column labels, column data types, memory usage, range index, and the number of cells in each column (non-null values).

In [8]:
tmdb_movies.describe()


id    popularity        budget       revenue       runtime  \
count   10866.000000  10866.000000  1.086600e+04  1.086600e+04  10866.000000   
mean    66064.177434      0.646441  1.462570e+07  3.982332e+07    102.070863   
std     92130.136561      1.000185  3.091321e+07  1.170035e+08     31.381405   
min         5.000000      0.000065  0.000000e+00  0.000000e+00      0.000000   
25%     10596.250000      0.207583  0.000000e+00  0.000000e+00     90.000000   
50%     20669.000000      0.383856  0.000000e+00  0.000000e+00     99.000000   
75%     75610.000000      0.713817  1.500000e+07  2.400000e+07    111.000000   
max    417859.000000     32.985763  4.250000e+08  2.781506e+09    900.000000   

         vote_count  vote_average  release_year    budget_adj   revenue_adj  
count  10866.000000  10866.000000  10866.000000  1.086600e+04  1.086600e+04  
mean     217.389748      5.974922   2001.322658  1.755104e+07  5.136436e+07  
std      575.619058      0.935142     12.812941  3.430616e+07  1.446325e+08  
min       10.000000      1.500000   1960.000000  0.000000e+00  0.000000e+00  
25%       17.000000      5.400000   1995.000000  0.000000e+00  0.000000e+00  
50%       38.000000      6.000000   2006.000000  0.000000e+00  0.000000e+00  
75%      145.750000      6.600000   2011.000000  2.085325e+07  3.369710e+07  
max     9767.000000      9.200000   2015.000000  4.250000e+08  2.827124e+09

 this demonstrates the  description of the data in the DataFrame for each column. 

In [9]:
tmdb_movies.isnull().sum()

id                         0
imdb_id                   10
popularity                 0
budget                     0
revenue                    0
original_title             0
cast                      76
homepage                7930
director                  44
tagline                 2824
keywords                1493
overview                   4
runtime                    0
genres                    23
production_companies    1030
release_date               0
vote_count                 0
vote_average               0
release_year               0
budget_adj                 0
revenue_adj                0
dtype: int64

The isnull function determines entries that are blank and without values in the dataset. 
in this case the cast of 76 movies has not been filled and the homepage pof 7930 movies has not been filled either.

In [10]:
tmdb_movies.isnull().sum() * 100 / len(tmdb_movies)

id                       0.000000
imdb_id                  0.092030
popularity               0.000000
budget                   0.000000
revenue                  0.000000
original_title           0.000000
cast                     0.699429
homepage                72.979937
director                 0.404933
tagline                 25.989324
keywords                13.740107
overview                 0.036812
runtime                  0.000000
genres                   0.211669
production_companies     9.479109
release_date             0.000000
vote_count               0.000000
vote_average             0.000000
release_year             0.000000
budget_adj               0.000000
revenue_adj              0.000000
dtype: float64

In this scenario the homepage column has a very high percentage of null values. Therefore, it is advised to drop the column since it has little data to draw conclusions from and is barely sgnificant in the analysis.

In [11]:
tmdb_movies.drop("homepage", axis= 1, inplace= True)

for the remaining columns drop the null values

In [12]:
#this is used to drop all the null values in the dataset
tmdb_movies.dropna(inplace = True)

In [13]:
tmdb_movies.isna().sum()

id                      0
imdb_id                 0
popularity              0
budget                  0
revenue                 0
original_title          0
cast                    0
director                0
tagline                 0
keywords                0
overview                0
runtime                 0
genres                  0
production_companies    0
release_date            0
vote_count              0
vote_average            0
release_year            0
budget_adj              0
revenue_adj             0
dtype: int64

In [14]:
#ax = sns.barplot(x= "bud", y = "popularity", data =tmdb_movies)

In [15]:
tmdb_movies['genres'].iloc[:20]

0     Action|Adventure|Science Fiction|Thriller
1     Action|Adventure|Science Fiction|Thriller
2            Adventure|Science Fiction|Thriller
3      Action|Adventure|Science Fiction|Fantasy
4                         Action|Crime|Thriller
5              Western|Drama|Adventure|Thriller
6     Science Fiction|Action|Thriller|Adventure
7               Drama|Adventure|Science Fiction
8             Family|Animation|Adventure|Comedy
9                       Comedy|Animation|Family
10                       Action|Adventure|Crime
11     Science Fiction|Fantasy|Action|Adventure
12                        Drama|Science Fiction
13                Action|Comedy|Science Fiction
14             Action|Adventure|Science Fiction
15                  Crime|Drama|Mystery|Western
16                        Crime|Action|Thriller
17             Science Fiction|Action|Adventure
18                 Romance|Fantasy|Family|Drama
19                War|Adventure|Science Fiction
Name: genres, dtype: object

In [16]:
tmdb_movies["popularity"].loc[:20]

0     32.985763
1     28.419936
2     13.112507
3     11.173104
4      9.335014
5      9.110700
6      8.654359
7      7.667400
8      7.404165
9      6.326804
10     6.200282
11     6.189369
12     6.118847
13     5.984995
14     5.944927
15     5.898400
16     5.749758
17     5.573184
18     5.556818
19     5.476958
20     5.462138
Name: popularity, dtype: float64

In [19]:
tmdb_movies["bud"].loc[:20]

KeyError: 'bud'